src/bitcoin-cli.cpp

```diff
@@ -747,8 +747,35 @@ static UniValue CallRPC(BaseRequestHandler* rh, const std::string& strMethod, co
     //     2. port in -rpcconnect (ie following : in ipv4 or ]: in ipv6)
     //     3. default port for chain
     uint16_t port{BaseParams().RPCPort()};
-    SplitHostPort(gArgs.GetArg("-rpcconnect", DEFAULT_RPCCONNECT), port, host);
-    port = static_cast<uint16_t>(gArgs.GetIntArg("-rpcport", port));
+    {
+        uint16_t rpcconnect_port{0};
+        const std::string rpcconnect_str = gArgs.GetArg("-rpcconnect", DEFAULT_RPCCONNECT);
+        if (!SplitHostPort(rpcconnect_str, rpcconnect_port, host)) {
+            throw std::runtime_error(strprintf("Invalid port provided in -rpcconnect: %s", rpcconnect_str));
+        } else {
+            if (rpcconnect_port != 0) {
```

Comment:
Interesting to find out that why the use of port number 0 is discouraged: https://www.lifewire.com/port-0-in-tcp-and-udp-818145

Reply:
Interesting article. Wonder what the behavior is before this PR when specifying 0..

Reply:
```bash
$ src/bitcoin-cli -rpcport=0 getblockchaininfo
error: timeout on transient error: Could not connect to the server 127.0.0.1:0

Make sure the bitcoind server is running and that you are connecting to the correct RPC port.

$ src/bitcoin-cli -rpcconnect=127.0.0.1:0 getblockchaininfo
error: timeout on transient error: Could not connect to the server 127.0.0.1:0

Make sure the bitcoind server is running and that you are connecting to the correct RPC port.
```

The client tries to reach port 0, which wouldn't be successful.

Reply:
Ah yeah, makes sense for the client. 👍